### PR TITLE
fix(tooltip): remove native event listener on component destroy

### LINKED ISSUE #4499: Tooltip Memory Leak
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
For tooltip to not leak memory.

#### What is the current behavior?
Tooltip seems to leak memory.

#### What are the steps to reproduce?
**Plunker with tooltip:**
https://plnkr.co/edit/IFpa3W9EXGG1Mch7TBmm?p=preview

**Plunker without tooltip**
https://plnkr.co/edit/vFdyEmedUYGCavMl8eEN?p=preview

#### Which versions of Angular, Material, OS, browsers are affected?
Used plunker template provided in issue template, so latest.

#### Is there anything else we should know?
I performed a devtools timeline recording and tried to keep it apples-to-apples. I performed GC at the start and end of each recording.

Strangely, both recordings seem to indicate leaking DOM nodes, which aren't cleaned up after a GC. The version with the tooltip however, doesn't clean up after itself nearly as nicely as the version without it.

**Timeline Without Tooltip**
without-tooltip

**Timeline With Tooltip**
with-toolitp

### PR DESCRIPTION
**Pull request by michastreppel**
Component should clean up events it has registered to.

Closes #4499

### CODE REVIEW

**Comment by michastreppel** [comment]
cc @andrewseguin

**Comment by Eddman** [comment]
Please merge ASAP: it is causing huge mem-leaks in my application so it makes the Chrome tab to crash in few minutes...

File: src/lib/tooltip/tooltip.ts [hunk 1]
```diff
@@ -185,8 +188,12 @@ export class MdTooltip implements OnDestroy {
...
     // they can prevent the first tap from firing its click event.
     if (!_platform.IOS) {
-      _renderer.listen(_elementRef.nativeElement, 'mouseenter', () => this.show());
-      _renderer.listen(_elementRef.nativeElement, 'mouseleave', () => this.hide());
+      this._enterListener = _renderer.listen(_elementRef.nativeElement,
+                                              'mouseenter',
+                                              () => this.show());
+      this._leaveListener = _renderer.listen(_elementRef.nativeElement,
+                                              'mouseleave',
+                                              () => this.hide());
```
Comment (jelbourn) [inline src/lib/tooltip/tooltip.ts]: Prefer breaking at the higher syntactic level:
```ts
    this._enterListener = 
        _renderer.listen(_elementRef.nativeElement, 'mouseenter', () => this.show());
    this._leaveListener = 
        _renderer.listen(_elementRef.nativeElement, 'mouseleave', () => this.hide());
```

**Comment by michastreppel** [comment]
@jelbourn I've resolved the comment and rebased the commit

**Comment by jelbourn** [comment]
@michastreppel please revert all of the formatting changes to the file

**Comment by michastreppel** [comment]
@jelbourn Done

**Comment by michastreppel** [comment]
@jelbourn @tinayuangao I've rebased this PR.

**Comment by DBosley** [comment]
Please merge this pr asap. The tooltips are causing major issues in my app.

**Comment by Emanjonil** [comment]
This is causing a major memory leak in my application. So if it could be fixed soon it would be much appreciated.

**Pull request merged**